aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -171,21 +171,29 @@ void magmaSolveBatched<float>(
 }
 
 template<>
-void magmaLu<double>(
+void cu_xgetrf<double>(
     magma_int_t m, magma_int_t n, double* dA, magma_int_t ldda,
     magma_int_t* ipiv, magma_int_t* info) {
-  MagmaStreamSyncGuard guard;
-  magma_dgetrf_gpu(m, n, dA, ldda, ipiv, info);
-  AT_CUDA_CHECK(cudaGetLastError());
+  auto handle = at::cuda::getCurrentCUDASolverDnHandle();
+  int lwork;
+  TORCH_CUSOLVER_CHECK(cusolverDnDgetrf_bufferSize(handle, m, n, dA, ldda, &lwork));
+  Tensor buffer = at::empty({lwork}, at::device(at::kCUDA).dtype(at::kDouble));
+  Tensor devInfo = at::empty({1}, at::device(at::kCUDA).dtype(at::kInt));
+  TORCH_CUSOLVER_CHECK(cusolverDnDgetrf(handle, m, n, dA, ldda, buffer.data_ptr<double>(), ipiv, devInfo.data_ptr<int>()));
```

Comment:
Can't we split this into two lines:
```
auto err = cusolverDnDngetrf(....)
TORCH_CUSOLVER_CHECK(err)
```

Reply:
I think basically all TORCH_*_CHECK macros are used inline.